Assets/Shopify/Plugins/iOS/Shopify/ApplePayButtonGenerator.mm

```diff
@@ -7,41 +6,40 @@
 extern "C" {
 #endif
     PKPaymentButtonStyle SBPayButtonStyleFromString(const char *style) {
-        if (strcmp(style, "white") == 0) {
+        if (strncmp(style, "WHITE", 5) == 0) {
             return PKPaymentButtonStyleWhite;
-        } else if (strcmp(style, "black") == 0) {
+        } else if (strncmp(style, "BLACK", 5) == 0) {
             return PKPaymentButtonStyleBlack;
-        } else if (strcmp(style, "whiteOutline") == 0) {
+        } else if (strncmp(style, "WHITE_OUTLINE", 11) == 0) {
             return PKPaymentButtonStyleWhiteOutline;
         } else {
             return PKPaymentButtonStyleBlack;
         }
     }
     
     PKPaymentButtonType SBPayButtonTypeFromString(const char *type) {
-        if (strcmp(type, "plain") == 0) {
+        if (strncmp(type, "PLAIN", 5) == 0) {
             return PKPaymentButtonTypePlain;
-        } else if (strcmp(type, "buy") == 0) {
+        } else if (strncmp(type, "BUY", 3) == 0) {
             return PKPaymentButtonTypeBuy;
-        } else if (strcmp(type, "setUp") == 0) {
+        } else if (strncmp(type, "SETUP", 5) == 0) {
             return PKPaymentButtonTypeSetUp;
-        } else if (strcmp(type, "inStore") == 0) {
+        } else if (strncmp(type, "IN_STORE", 6) == 0) {
```

Comment:
Should this be a length of 7? This seems kind of error prone, perhaps the constants could be declared like so?
```C
char plainType[] = "PLAIN";
strncmp(type, plainType, sizeof(plain));
```